packages/contracts-bedrock/src/universal/OptimismMintableERC20FactoryInterop.sol

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { OptimismMintableERC20Factory } from "./OptimismMintableERC20Factory.sol";
+import { Predeploys } from "src/libraries/Predeploys.sol";
+import { Unauthorized } from "src/libraries/errors/CommonErrors.sol";
+
+/// @custom:proxied true
+/// @custom:predeployed 0x4200000000000000000000000000000000000012
+/// @title OptimismMintableERC20FactoryInterop
+/// @notice OptimismMintableERC20FactoryInterop is an extension of the OptimismMintableERC20Factory that handles the
+/// `deployments` mapping inclusion of the tokens deployed before holocene. It uses a hash onion structure to verify and
+/// store the deployments, and can be consired complete when the hash onion is fully peeled.
+contract OptimismMintableERC20FactoryInterop is OptimismMintableERC20Factory {
+    /// @notice Thwown when the hashOnion is already set.
+    error HashOnionAlreadySet();
+
+    /// @notice Thrown when the hashOnion is not set and `verifyAndStore` is called.
+    error HashOnionNotSet();
```

Comment:
I wonder if we should add this to the abi considering it should never throw. If no one set the hashonion, the storage will be at 0. I dare you to find a set of multiple values such that the hash of the concatenation is equal to 0. It will fail to verify.